src/work-allocation/components/task-assignment/task-assignment.component.ts

```diff
@@ -135,18 +196,30 @@ export class TaskAssignmentComponent implements OnInit {
       });
     } else if (this.location && this.location.id) {
       // Otherwise, get the caseworkers at the specifed location.
-      this.caseworkerService.getForLocation(this.location.id).subscribe(caseworkers => {
-        this.setupCaseworkers(caseworkers);
+      this.caseworkerService.getAll().subscribe(caseworkers => {
+        const locationCaseWorkers = caseworkers.filter(cw => this.locationMatchesSelectedLocation(cw));
+        this.setupCaseworkers(locationCaseWorkers);
       }, error => {
         handleFatalErrors(error.status, this.router, WILDCARD_SERVICE_DOWN);
       });
     }
   }
 
-  // Sets up the caseworkers, ensuring that excluded ones that are filtered out.
+  /**
+   * Checks whether the caseworker's location matches the location selected in the dropdown
+   * @param cw The caseworker to consider.
+   */
+  private locationMatchesSelectedLocation(cw: Caseworker): boolean {
+    return cw.location ? cw.location.id.toString() === this.location.id : false;
+  }
+
+  /**
+   * Sets up the caseworkers, ensuring that excluded ones that are filtered out.
+   * @param caseworkers The initial list of caseworkers
+   */
   private setupCaseworkers(caseworkers: Caseworker[]): void {
-    this.pAllCaseworkers = [ ...caseworkers ];
-    this.pCaseworkers = [ ...caseworkers ].filter(item => {
+    this.pAllCaseworkers = [...caseworkers];
+    this.pCaseworkers = [...caseworkers].filter(item => {
```

Comment:
[...caseworkers].filter() is this necessary filter creates a new array anyway.

Reply:
I did not write this code (just reformatted) however could just change [...caseworkers].filter -> pAllCaseworkers.filter if that's suitable Kuda?

Reply:
If this is possible, yes please.

Reply:
Ok, will do